lib/beaker/host.rb

```diff
@@ -294,9 +294,25 @@ def is_x86_64?
 
     def connection
       # create new connection object if necessary
+      # Pull the connection information from the host configuration
+      ssh_opts = self['ssh']
+      ssh_opts = {} unless ssh_opts.is_a?(Hash)
+
+      if self.options['ssh']
+        if ssh_opts[:config]
+          # Load the configuration from the SSH config file
+          ssh_opts = self.options['ssh'].merge(Net::SSH::Config.load(ssh_opts[:config], self['ip']))
+        else
+          ssh_opts = self.options['ssh']
```

Comment:
I'm trying to understand this code path. Before this PR, the connection would always use `self['ssh']` (hardcoded). With this PR, when `self.options['ssh']` is present AND `[:config:]` is false (default unless specifying a config file) we now use this line `ssh_opts = self.options['ssh']`. The end result is hardcoding `self.options['ssh']` and the previous behaviour of having `self['ssh']` is lost. I think this is the issue we are hitting in our acceptance tests because we don't specify external ssh config files.

Reply:
Hmm...interesting.

We also don't use external SSH config files but Vagrant does so we had to figure out how to manipulate that.

Is it possible that we were just trying to specify the SSH option incorrectly?

Basically, we just need to change the SSH session timeout option for long running jobs and couldn't figure out how to do that successfully without modifying the underlying Vagrant SSH configuration file.